### ISSUE TITLE
v1.0.0: latest is broken in windows

### ISSUE THREAD

**Issue by husayt** [module-federation/vite#68]
Examples stopped working for me with latest changes (on windows with `dev` run):

I get both host and remote broken with this error
```
chunk-R27WTGMQ.js:21  Uncaught TypeError: factory is not a function
    at chunk-R27WTGMQ.js:21:29
    at async chunk-R27WTGMQ.js:27:1
```

error happens here:
```js
 var exportModule = (
      /*mf top-level-await placeholder replacement mf*/
      res.then((factory) => factory())
    );
``` 
`factory` comes out as `false`, hence `factory is not a function` error

this is wider code of where error happens
```
// node_modules/__mf__virtual/__loadShare__react__$2__jsx__$3__dev__$3__runtime.js
var require_loadShare_react_2_jsx_3_dev_3_runtime = __commonJS({
  "node_modules/__mf__virtual/__loadShare__react__$2__jsx__$3__dev__$3__runtime.js"(exports, module) {
    var { loadShare } = require_index_cjs();
    var res = loadShare("react/jsx-dev-runtime", {
      customShareInfo: { shareConfig: {
        singleton: false,
        strictVersion: false,
        requiredVersion: "^18.3.1"
      } }
    });
    var exportModule = (
      /*mf top-level-await placeholder replacement mf*/
      res.then((factory) => factory())
    );
    module.exports = exportModule;
  }
});
```

**Comment by zhangHongEn** [comment]
I rented a cloud service windows server, but I haven't configured it successfully yet

**Comment by zhangHongEn** [comment]
Can try it first
`server.watch.usePolling:true`

**Comment by zhangHongEn** [comment]
https://vitejs.dev/config/server-options.html#server-watch

**Comment by husayt** [comment]
I tried that, but still the same.

**Comment by zhangHongEn** [comment]
https://github.com/vitejs/vite/issues/8619
Found the problem, but not the solution yet

**Comment by gioboa** [comment]
Closed by #71
